### PR TITLE
[OcSelect] Fix search for options provided as objects

### PR DESCRIPTION
**Pull request by dschmidt**
## Description
I made `OcSelect.filter`make use of the `label` prop as the original filter function of vue-select also does, i.e. this is a regression fix for a breaking change introduced in https://github.com/owncloud/owncloud-design-system/pull/1521

## Related Issue

- Fixes <issue_link>

## Motivation and Context
This fixes broken search in ownBrander 2.0.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation added/updated

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by fschade** [comment]
@dschmidt can you please have a look if this works for you

**Comment by dschmidt** [comment]
Yep, works for me.

Feel free to merge if you're happy now (I just cannot approve my own PR)

File: src/components/table/OcTableFiles.spec.js [hunk 1]
```diff
@@ -179,19 +179,19 @@ describe("OcTableFiles", () => {
     })
   })
 
-  describe("resource details", () => {
+  test.skip("resource details", () => {
```
Comment (kulmann) [inline src/components/table/OcTableFiles.spec.js:182]: `test.skip`? 🤔

Comment (kulmann) [reply src/components/table/OcTableFiles.spec.js:182]: ah, because they were not enabled before

**Review by kulmann — APPROVED** [review]

**Comment by dschmidt** [comment]
🎉

**Pull request merged**